deepreg/model/layer.py

```diff
@@ -425,25 +427,31 @@ def __init__(
             0.44 * cp for cp in control_point_spacing
         ]  # 0.44 = ln(4)/pi
         self.cp_spacing = control_point_spacing
-        self.kernel = None
+        self.kernel = [gaussian_kernel1d(sigma=sigma) for sigma in self.kernel_sigma]
         self._output_shape = None
         self._resize = None
 
     def build(self, input_shape):
         super().build(input_shape=input_shape)
 
-        self.kernel = layer_util.gaussian_filter_3d(self.kernel_sigma)
         output_shape = tuple(
-            tf.cast(tf.math.ceil(v / c) + 3, tf.int32)
-            for v, c in zip(input_shape[1:-1], self.cp_spacing)
+            [

```

Comment:
no need of `[ ]` here, `tuple(x for x in xx)` is fine, so this line does not need to be modified